pkg/model/components/kindnet.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package components
+
+import (
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/loader"
+)
+
+// KindnetOptionsBuilder adds options for kindnet to the model
+type KindnetOptionsBuilder struct {
+	Context *OptionsContext
+}
+
+var _ loader.ClusterOptionsBuilder = &KindnetOptionsBuilder{}
+
+func (b *KindnetOptionsBuilder) BuildOptions(o *kops.Cluster) error {
+	clusterSpec := &o.Spec
+	c := clusterSpec.Networking.Kindnet
+	if c == nil {
+		return nil
+	}
+
+	if c.Version == "" {
+		c.Version = "stable"
```

Comment:
Is there an immutable tag we can use here? That way kindnet upgrades are tied to kops operations and not replacement nodes pulling new versions of a mutable tag.

Reply:
fair enough, let me change it